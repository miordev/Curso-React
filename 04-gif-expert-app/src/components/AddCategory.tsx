import { useState } from 'react';

export const AddCategory = ({ onNewCategory }: any) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    setInputValue(event.target.value);
  };

  const onInputSubmit = (event: React.FormEvent<HTMLFormElement>): void => {
    event.preventDefault();

    if (inputValue.trim().length > 1) {
      onNewCategory(inputValue.trim());
      setInputValue('');
    }
  };

  return (
    <form onSubmit={onInputSubmit}>
      <input type="text" placeholder="Search Gifs" value={inputValue} onChange={onInputChange} />
    </form>
  );
};
