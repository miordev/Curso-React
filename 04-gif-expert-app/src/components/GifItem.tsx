import PropTypes from 'prop-types';

export const GifItem = ({ url, title }: any) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p data-testid="test-title">{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
