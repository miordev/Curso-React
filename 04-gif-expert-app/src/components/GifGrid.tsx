import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';
import { GifItem } from './GifItem';

export const GifGrid = ({ category }: any) => {
  const [gifList, setGifList] = useState<any[]>([]);

  const updateGifList = async () => {
    const newGifList = await getGifs(category);
    setGifList(newGifList);
  };

  useEffect(() => {
    updateGifList();
  }, []);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {gifList.map((gif: any) => (
          <GifItem key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};
