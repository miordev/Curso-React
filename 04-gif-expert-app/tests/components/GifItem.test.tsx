import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Tests in <GifItem/>', () => {
  const mockTitle = 'Goku';
  const mockUrl = 'https://goku.jpg/';

  test('should do a match with the snapshot', () => {
    const { container } = render(<GifItem title={mockTitle} url={mockUrl} />);
    expect(container).toMatchSnapshot();
  });

  test('should show the title', () => {
    render(<GifItem title={mockTitle} url={mockUrl} />);
    expect(screen.getByTestId('test-title').innerHTML).toBe(mockTitle);
  });

  test('should show the image', () => {
    render(<GifItem title={mockTitle} url={mockUrl} />);
    const { src, alt } = screen.getByRole('img') as HTMLImageElement;
    expect(src).toBe(mockUrl);
    expect(alt).toBe(mockTitle);
  });
});
