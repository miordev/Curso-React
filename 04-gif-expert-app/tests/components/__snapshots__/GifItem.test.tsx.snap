// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifItem/> should do a match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://goku.jpg/"
    />
    <p
      data-testid="test-title"
    >
      Goku
    </p>
  </div>
</div>
`;
